youtube_dl/extractor/bbc.py

```diff
@@ -959,6 +959,32 @@ def _real_extract(self, url):
              r'videoId\s*:\s*["\'](%s)["\']' % self._ID_REGEX],
             webpage, 'vpid', default=None)
 
+        # bbc reel (e.g. https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness)
+        if not programme_id:
+            initial_data = self._search_regex(
+                r'<script[^>]+id="initial-data"[^>]+data-json=\'(.+)\'>',
+                webpage, 'initial data', default=None)
+            if initial_data:
+                # let's see if it actually is, or ExtractorError
+                initial_data = self._parse_json(unescapeHTML(initial_data), 'initial data')
+
+                def _extract_pid(data):
+                    if isinstance(data, dict):
+                        if data.get('kind', '') == 'programme' and 'versionID' in data:
+                            return data['versionID']
+                        else:
+                            for k in data:
+                                pid = _extract_pid(data[k])
+                                if pid:
+                                    return pid
+                    elif isinstance(data, list):
+                        for v in data:
+                            pid = _extract_pid(v)
+                            if pid:
+                                return pid
+                    return None
+                programme_id = _extract_pid(initial_data)
```

Comment:
this is too generic and can lead to picking the wrong video.

Reply:
Thanks, as far as I can see, we need the `clipPID` from the first element of the JSON, whose `items` seems to be a playlist starting at the current video. ajj8 went for `['smpData']['items'][0]['versionID'].

https://github.com/dirkf/youtube-dl-1/commit/35bb60e7488ae73db6c49737aaf98824086f1108

I've actually watched https://www.bbc.com/reel/video/p0931jtk/the-nuclear-lighthouses-built-by-the-soviets-in-the-arctic, downloaded with the code of the commit above and it matches the `['smpData']['items'][0]['duration']: I guess the `versionID` and `clipPID` both resolve to the same formats page. 

```
# ffprobe -hide_banner "The_nuclear_lighthouses_built_by_the_Soviets_in_the_Arctic_-_BBC_Reel.mp4"
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/media/drive1/Video The_nuclear_lighthouses_built_by_the_Soviets_in_the_Arctic_-_BBC_Reel.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.20.100
  Duration: 00:05:24.48, start: 0.000000, bitrate: 831 kb/s
    Stream #0:0(und): Video: h264 (Main) (avc1 / 0x31637661), yuv420p(tv, bt709), 704x396 [SAR 1:1 DAR 16:9], 730 kb/s, 25 fps, 25 tbr, 90k tbn, 50 tbc (default)
    Metadata:
      handler_name    : VideoHandler
    Stream #0:1(eng): Audio: aac (HE-AAC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 94 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
```

Again, as the pid is in the URI, that might have been an easier target.

Reply:
On further investigation `versionID` is definitely the correct PID.

The implementation in https://github.com/ytdl-org/youtube-dl/pull/28268/commits/db1096f4b92fbf1be6f8393f46b034f8bc09f8bb goes for `initData['items'][0]`. If that is a `dict` and has `smpData` and that in turn has `['items'][0]` as a `dict`, the `smpData` and its child `dict` are merged into the parent `dict` from which the metadata is pulled.